src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx

```diff
@@ -0,0 +1,82 @@
+import { Suspense, useState } from 'react';
+
+import Box from '@mui/material/Box';
+import CircularProgress from '@mui/material/CircularProgress';
+import { Container } from '@mui/system';
+import { Await, useLoaderData } from 'react-router-dom';
+import { style } from 'typestyle';
+
+import { useAppSelector } from '../../../hooks/app';
+import { Colors, Spacing } from '../../../styles';
+import type { SubtestsRevisionsHeader } from '../../../types/state';
+import DownloadButton from '.././DownloadButton';
+import type { LoaderReturnValue } from '.././loader';
+// import RevisionSelect from '.././RevisionSelect';
+import SearchInput from '.././SearchInput';
+import SubtestsResultsTable from './SubtestsResultsTable';
+import SubtestsRevisionHeader from './SubtestsRevisionHeader';
+
+function SubtestsResultsMain() {
+  const { results } = useLoaderData() as LoaderReturnValue;
+
+  const themeMode = useAppSelector((state) => state.theme.mode);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const subtestsHeader: SubtestsRevisionsHeader = {
```

Comment:
I have a typescript error here.

Reply:
For the typescript error, it might get fixed if you get `results` data directly in `SubtestsRevisionHeader` instead. This assumes you've deferred the returned data from the loaders as mentioned in my comment below. 

See here: [Await component docs](https://reactrouter.com/en/main/components/await) and here: [UseAsyncValue docs](https://reactrouter.com/en/main/hooks/use-async-value)

A suggestion:


```
import { useAsyncValue } from 'react-router-dom';
....

function SubtestsRevisionHeader() {
  const loaderData = useAsyncValue();
  const resultsData = loaderData as CompareResultsItem[][];

 //filter results if needed

  const results = filteredResults as CompareResults[];


  const subtestsHeader: SubtestsRevisionsHeader = {
    suite: results[0].suite,
    framework_id: results[0].framework_id,
    test: results[0].test,
    option_name: results[0].option_name,
    extra_options: results[0].extra_options,
    new_rev: results[0].new_rev,
    new_repo: results[0].new_repository_name,
    platform: results[0].platform,
  };

....
```